type of literal array index

The following program `sample.go` triggers a panic:

```go
package main

import "fmt"

type rule uint8

const (
    r0 rule = iota
    r1
    r2
)

var a = [...]int{
    r0: 1,
    r1: 12,
}
    
func main() {
    fmt.Println(a)
}
```

Expected result:
```console
$ go run ./sample.go
[1 12]
```

Got:
```console
$ yaegi ./sample.go
panic: reflect: call of reflect.Value.Int on uint8 Value

goroutine 1 [running]:
reflect.Value.Int(...)
        /usr/lib/go/src/reflect/value.go:986
github.com/containous/yaegi/interp.arrayLit(0xc0000efc00)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:1465 +0x660
github.com/containous/yaegi/interp.setExec.func1(0xc0000efc00)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1832 +0xb2
github.com/containous/yaegi/interp.setExec.func1(0xc000260300)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1821 +0x25d
github.com/containous/yaegi/interp.setExec.func1(0xc000260000)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1821 +0x25d
github.com/containous/yaegi/interp.setExec.func1(0xc0000eec00)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1821 +0x25d
github.com/containous/yaegi/interp.setExec.func1(0xc0000ef500)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1821 +0x25d
github.com/containous/yaegi/interp.setExec.func1(0xc0000ef100)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1821 +0x25d
github.com/containous/yaegi/interp.setExec.func1(0xc0000eed00)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1821 +0x25d
github.com/containous/yaegi/interp.setExec(0xc0000eed00)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1835 +0xf5
github.com/containous/yaegi/interp.genRun.func1(0xc0000eec00, 0xc0001a3d48)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1562 +0x44
github.com/containous/yaegi/interp.(*node).Walk(0xc0000eec00, 0xc0001a3d48, 0x0)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:164 +0xc9
github.com/containous/yaegi/interp.(*node).Walk(0xc0000ee300, 0xc0001a3d48, 0x0)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:168 +0x70
github.com/containous/yaegi/interp.genRun(0xc0000ee300, 0xc00023c240, 0xc0000c4a08)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1550 +0x64
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc000240dc0, 0xc00025e000, 0x96, 0x296, 0xc00025e000, 0x96, 0x0, 0x1)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:353 +0x388
main.main()
        /home/marc/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:82 +0x42e
```

--- a/_test/a40.go
+++ b/_test/a40.go
@@ -1,0 +1,23 @@
+package main
+
+import "fmt"
+
+type rule uint8
+
+const (
+	r0 rule = iota
+	r1
+	r2
+)
+
+var a = [...]int{
+	r0: 1,
+	r1: 12,
+}
+
+func main() {
+	fmt.Println(a)
+}
+
+// Output:
+// [1 12]

--- a/interp/run.go
+++ b/interp/run.go
@@ -1461,7 +1461,7 @@ func arrayLit(n *node) {
 		if c.kind == keyValueExpr {
 			convertLiteralValue(c.child[1], rtype)
 			values[i] = genValue(c.child[1])
-			index[i] = int(c.child[0].rval.Int())
+			index[i] = int(vInt(c.child[0].rval))
 		} else {
 			convertLiteralValue(c, rtype)
 			values[i] = genValue(c)

